content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+- Style composition, I believe that regular Javascript object composition is better than CSS cascading global composition model, plus you can create a custom system that fits your team and your business logic, remember, it's regular Javascript. You can easily publish your Themes as private NPM packages internally to shared them among Applications.
+
+While regular inline-styles will go a long way, you will most likely
+need features that inline-styles do not provide and that regular CSS do such
+as pseudo selectors like `:hover` and media queries.
+To solve this problem and provide better Developer Experience overall is that
+**CSS-in-JS** exist.
+
+I recommend going for [Emotion][5] or [Styled-components][6] since they
+are very mature and feature rich, have
+good documentation and an active community.
+
+#### As Component Consumers
+
+Here is where the **Inline** approach _shines_.
+To use a module with all its styles and assets such as images
+you simply need to write a single line of code:
+
+```javascript
+import module from "module"
+```
+
+That is all!
+
+If you want to import the module dynamically (to parallelize and deffer its loading
+in the client) the experience is very similar:
+
+```javascript
+// Inside some function or component
+const module = await import("module")
+```
+
+If you have Webpack configured correctly to create chunks that can be
+dynamically loaded, this is called `Code Splitting`, then `module`
+will be an independent file lazy loaded, only and when its needed.
+
+Latest version of Webpack make this configuration almost trivial,
+check out their [docs][110].
+
+This approach also works with the _Classic Way_ of using dependencies via Browser Globals.
+
+## Active vs Passive compilation
+
+> **TLDR: Do not use _Passively compiled_ dependecies ever!**
+
+I have seen situations in which packages are distributed raw,
+without being compiled before being published, this means **Component Authors**
+package and distribute the raw sources like Typescript, ES2054, JSX, CSS Modules, etc.,
+and rely on **Component Consumers**, like Applications, to compile them usually
+by configuring Webpack to also treat `node_modules/my-raw-dependency`
+as part of the Application source.
+
+This is what I call **Passive compilation** because **Component Authors**
+delegate their package compilation to the **Component Consumers**, instead
+of **Active compilation** in which each package is distributed already compiled
+by its **Component Author**.
+
+Today you can go a long way with Webpack abstractions like
+[Create React App][16] and [Create React Library][105], but you need to
+do things in a fairly standard way and **Passive Compilation** is not at all
+a standard way of distributing code.
+
+In my experience Webpack configurations are something that most developers
+do not know or do not want to touch so
+**having these sort of abstractions inside
```

Comment:
_These sorts_ or _this sort_, because the first word refers directly to the second one so it depends on the noun which in the first case is plural and in the second one, singular.